src/App.js

```diff
@@ -1,12 +1,122 @@
 import './App.css';
 import firebase from './firebase';
+import { useEffect, useState, useRef } from 'react';
+import { getDatabase, ref, onValue, push, remove } from 'firebase/database';
+import LoggedEntries from './LoggedEntries';
 
 function App() {
+  const [entry, setEntry] = useState([]);
+  const [userEntry, setUserEntry] = useState("");
+  const scrollToEntry = useRef();
+
+  useEffect(() => {
+    const database = getDatabase(firebase)
+    const dbRef = ref(database)
+
+    onValue(dbRef, (response) => {
+      const newState = [];
+      const data = response.val();
+      
+      for (let key in data) {
+        newState.push(
+          {
+            key: key,
+            name: data[key]
+          });
+      }
+      setEntry(newState);
+    })
+
+  }, [])
+
+    const handleSubmit = (event) => {
+      event.preventDefault();
+      const database = getDatabase(firebase)
+      const dbRef = ref(database);
+      
+      push(dbRef, userEntry);
+      setUserEntry("");
+      scrollToEntry.current.scrollIntoView({ behavior: 'smooth' })
+    }
+
+    const handleInputChange = (event) => {
+      setUserEntry(event.target.value);
+    }
+
+    const handleBurn = (burnEntry) => {
+      const database = getDatabase(firebase)
+      const dbRef = ref(database, `${burnEntry}`);
+      remove(dbRef);
+    }
+  
   return (
-    <div className="App">
-      <header className="App-header">
+    <div className="App"> 
```

Comment:
Not necessary but if you wanted to break this down into further components you could always take this entire header and put it into a "stateless" component and just put <Header /> here, then that whole chunk of code would sort of be tucked out of the way